### PR TITLE
enh(datepicker): change start/end date/time when one of the dates is …

### LINKED ISSUE #3642: Context-aware help in Administration
---

Author Name: **Julien Mathis** (Julien Mathis)
Original Redmine Issue: 1995, https://forge.centreon.com/issues/1995
Original Date: 2010-10-14
Original Assignee: Sylvestre Ho

---

None

### PR DESCRIPTION
**Pull request by miteto**
<h1> Fix datepicker </h1>

<h2> change start/end date/time when one of the dates is changed </h2>

<h2> Type of change </h2>

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

<h2> Target serie </h2>

- [ ] 2.8.x
- [ ] 18.10.x
- [X] 19.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

<h2> Checklist </h2>

<h5> Community contributors & Centreon team </h5>

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

<h5> Centreon team only </h5>

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests covers 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/monitoring/downtime/template/AddDowntime.ihtml [hunk 1]
```diff
@@ -154,5 +154,26 @@
...
+        $(".timepicker").first().on('change', function (e) {
+            updateEndTime();
+        });
+        $("#duration").on('keypress', function (e) {
+            updateEndTime();
+        });
+        $("#duration_scale").on('change', function (e) {
+            updateEndTime();
+        });
+        $(".datepicker").last().on('change', function (e) {
```
Comment (sc979) [inline www/include/monitoring/downtime/template/AddDowntime.ihtml]: can you remove this part please.
I misunderstood what you meant when you told me "it needs to work both ways"
It's not both ways when changing start/end date. But both ways when changing start date/time

**Comment by sc979** [comment]
BTW : there's a class centreonDuration.class.php which do almost every duration modifications you added :'( but it works with seconds instead of hours

**Comment by sc979** [comment]
linked to these PR too :
https://github.com/centreon/centreon-widget-service-monitoring/pull/117
https://github.com/centreon/centreon-widget-host-monitoring/pull/65

File: www/include/monitoring/downtime/AddDowntime.php [hunk 2]
```diff
@@ -319,8 +320,16 @@
...
+        $defaultDuration *= 60;
+    } elseif ($defaultScale == 'h') {
+        $defaultDuration *= 3600;
+    } elseif ($defaultScale == 'd') {
+        $defaultDuration *= 86400;
+    }
     $data["start_time"] = $centreonGMT->getDate("G:i", time(), $gmt);
-    $data["end_time"] = $centreonGMT->getDate("G:i", time() + 7200, $gmt);
+    $data["end_time"] = $centreonGMT->getDate("G:i", time() + $defaultDuration, $gmt);
+    $data["end"] = $centreonGMT->getDate("m/d/Y", time() + $defaultDuration, $gmt);
```
Comment (loiclau) [inline www/include/monitoring/downtime/AddDowntime.php]: Need to check if the hard coded format works with all localization.

File: www/include/monitoring/downtime/template/AddDowntime.ihtml [hunk 3]
```diff
@@ -154,5 +154,14 @@
 
     initDatepicker("datepicker", "mm/dd/yy", "0");
 
+    $(document).ready(function() {
+        $(".datepicker").first().on('change', function (e) {
```
Comment (loiclau) [inline www/include/monitoring/downtime/template/AddDowntime.ihtml]: if possible use other locator of first or last.

File: www/include/monitoring/downtime/template/AddDowntime.ihtml [hunk 4]
```diff
@@ -154,5 +154,14 @@
...
+            updateEndTime();
+        });
+        $(".datepicker").last().on('change', function (e) {
+            updateStartTime();
+        });
+    });
```
Comment (loiclau) [inline www/include/monitoring/downtime/template/AddDowntime.ihtml]: make an update of date with the timepicker too

**Comment by loiclau** [comment]
Dont use duration for set the donwtime period, duration is only for flexible downtime and have link with the period.

File: www/include/monitoring/downtime/template/AddDowntime.ihtml (hunk 4 above)
Comment (miteto) [reply www/include/monitoring/downtime/template/AddDowntime.ihtml]: It is not described in the issue

**Pull request closed without merging**